components/list/list-item.js

```diff
@@ -0,0 +1,183 @@
+				max-width: 90px;
+				overflow:hidden;
+			}
+			:host([dir="rtl"]) ::slotted([slot|="illustration"]) {
+				margin-left: 0.9rem;
+				margin-right: 0rem;
+			}
+			::slotted([slot="illustration-outer"]) {
+				margin-bottom: 18px;
+				margin-top: 18px;
+			}
+			::slotted([slot="actions"]) {
+				align-self: flex-start;
+				display: flex;
+				flex-grow: 0;
+			}
+			.d2l-list-item-main {
+				width: 100%;
+			}
+		`;
+		const mainContent = css`
+			::slotted(.d2l-list-item-text) {
+				margin: 0;
+				max-height: 2.4rem;
+				overflow: hidden;
+			}
+			::slotted(.d2l-list-item-text-secondary-responsive),
+			::slotted(.d2l-list-item-text-secondary) {
+				margin: 0;
+				margin-top: 0.3rem;
+				overflow: hidden;
+			}
+			::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: none;
+			}
+		`;
+		const divider = css`
+			.d2l-list-item-divider-top,
+			.d2l-list-item-divider-bottom {
+				border: 0;
+				border-top: 1px solid var(--d2l-color-mica);
+				margin: 0;
+				position: absolute;
+				width: 100%;
+			}
+			.d2l-list-item-divider-top {
+				display: var(--d2l-list-item-divider-top, none);
+				top: -1px;
+			}
+			.d2l-list-item-divider-bottom {
+				bottom: -1px;
+				display: var(--d2l-list-item-divider-bottom, none);
+			}
+		`;
+
+		const breakPoint580 = css`
+			.d2l-list-item-flex[breakpoint="580"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 71px;
+				max-width: 120px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="580"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="580"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+
+		const breakPoint636 = css`
+			.d2l-list-item-flex[breakpoint="636"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 102px;
+				max-width: 180px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="636"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="636"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+
+		const breakPoint842 = css`
+			.d2l-list-item-flex[breakpoint="842"] ::slotted([slot|="illustration"]) {
+				margin-right: 1rem;
+				max-height: 120px;
+				max-width: 216px;
+			}
+			:host([dir="rtl"]) .d2l-list-item-flex[breakpoint="842"] ::slotted([slot|="illustration"]) {
+				margin-left: 1rem;
+				margin-right: 0;
+			}
+			.d2l-list-item-flex[breakpoint="842"] ::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: block;
+			}
+		`;
+		return [ layout, mainContent, divider, breakPoint580, breakPoint636, breakPoint842];
+	}
+
+	constructor() {
+		super();
+		this._breakpoint = 0;
+		this._breakpointList = [842, 636, 580, 0];
+	}
+
+	render() {
+		return html`
+			<div class="d2l-list-item-flex" breakpoint="${this._breakpoint}">
```

Comment:
Agreed, we'll want to be careful that we get the semantics right here and don't break any HTML markup rules (e.g. `ul` elements can only contain `li` elements).

Reply:
Suggestions then? 

I think I did forget to add the role on the `list-items`.  But given your comment about "(e.g. ul elements can only contain li elements)" makes me think I should change `ul`?

Reply:
I'd suggest doing more experiments. We definitely shouldn't be mixing and matching `<ul>`s with `<div>`s -- we should either be using `<ul>`s and `<li>`s or only `<div>`s with the correct roles on them. I have a feeling you're going to need the latter since you can't have `<ul><some-custom-element><li>` anyway.

Reply:
I agreed. Will go and fix this up.

Reply:
This has been fixed.